File: results/transcoder/translations/py/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.cpp
bool check ( string st ) {
  int n = st . length ( ) ;
  if ( ( n == 0 ) || ( n == 1 ) ) return false ;
  if ( ( n == 2 ) || ( n == 3 ) ) return ( ( st [ 0 ] - '0' ) % 4 == 0 ) ;
  int last = ( int ) ( st [ n - 1 ] ) ;
  int secondLast = ( int ) ( st [ n - 2 ] ) ;
  return ( ( secondLast * 10 + last ) % 4 == 0 ) ;
}

File: cfgs/java_cfgs_foo/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.java
Nodes
1;special;;;
2;statement;int n = fun_1();14;
3;conditional;foo_1 == 0;15;
4;statement;return false;15;
5;exit;;;
6;conditional;foo_1 == 1;16;
7;statement;return ((fun_1(0) - '0') % 4) == 0;16;
8;statement;int last = fun_1(foo_2 - 1) - '0';17;
9;statement;int second_last = fun_1(foo_2 - 2) - '0';18;
10;statement;return (((foo_1 * 10) + foo_2) % 4) == 0;19;
Edges
0;2;
1;3;
2;4;6;
3;5;
4;
5;7;8;
6;5;
7;9;
8;10;
9;5;
